testplan/web_ui/web_app.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+"""
+Web application for Testplan & Monitor UIs,
+"""
+from testplan.common.utils.path import pwd
+from flask import Flask, send_from_directory, abort
+from flask_restplus import Resource, Api
+from werkzeug.exceptions import NotFound, NotImplemented
+import argparse
+import os
+
+app = Flask(__name__)
+_api = Api(app)
+
+def parse_cli_args():
+    """Web App command line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--static-path', nargs='?', default=None, const=pwd())
+    parser.add_argument('--data-path', nargs='?', default=None, const=pwd())
+    return parser.parse_args()
+
+@_api.route('/testplan/<string:report_uid>')
+class Testplan(Resource):
+    def get(self, report_uid):
+        """Get a Testplan report (HTML) given it's uid."""
+        directory = os.path.join(app.config['STATIC_PATH'], 'testing')
+        return send_from_directory(directory=directory, filename='index.html')
+
+@_api.route('/testplan/<string:report_uid>/report')
+class TestplanReport(Resource):
+    def get(self, report_uid):
+        """Get a Testplan report (JSON) given it's uid."""
+        directory = os.path.join(
+            app.config['DATA_PATH'],
+            'reports',
+            'testplan_{uid}'.format(uid=report_uid)
+        )
+        return send_from_directory(directory=directory, filename='report.json')
+
+@_api.route('/testplan/<string:report_uid>/assertions/<string:assertions_uid>')
+class TestplanAssertions(Resource):
+    def get(self, report_uid, assertions_uid):
+        """Get an Assertion report (JSON) for a specific Testplan report given their uids."""
+        raise NotImplemented()  # pylint: disable=notimplemented-raised
```

Comment:
It's not great to shadow the NotImplemented builtin, would slightly prefer `from werkzeug import exceptions` and use as `exceptions.NotImplemented`.

Reply:
Fair point, I'll change it